server/src/main/java/org/opensearch/action/admin/cluster/remotestore/restore/RestoreRemoteStoreRequest.java

```diff
@@ -53,14 +55,18 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeStringArray(indices);
         out.writeOptionalBoolean(waitForCompletion);
         out.writeOptionalBoolean(restoreAllShards);
+        out.writeOptionalString(clusterUUID);
     }
 
     @Override
     public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = null;
-        if (indices == null || indices.length == 0) {
+        if ((indices == null || indices.length == 0) && clusterUUID.isEmpty() == true) {
             validationException = addValidationError("indices are missing", validationException);
         }
+        if (restoreAllShards == true && clusterUUID.isEmpty() == false) {
+            validationException = addValidationError("restoreAllShards cannot be true when clusterUUID is set", validationException);
```

Comment:
`restoreAllShards` should not be considered when `cluserUUID` is set, right? Why do we need to specifically check for `restoreAllShards == true`?

Reply:
added this check to through error if user is trying to send in restoreAllShards and clusterUUID both as restoreAllShards requires indices to be in the closed state. 

We can silently ignore the `restoreAllShards` flag if clusterUUID param exists. Let me know your thoughts

Reply:
Yes, I think we should ignore `restoreAllShards` if `clusterUUID` is present. We can clarify this in the documentation. The problem with current implementation is that if user explicitly passes `restoreAllShards=false`, we will not the validationException making the behavior inconsistent.